frontend/src/pages/BYONImages/BYONImageModal/HardwareProfileIdentifierMultiselect.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { useDashboardNamespace } from '~/redux/selectors';
+import { MultiSelection } from '~/components/MultiSelection';
+import useHardwareProfiles from '~/pages/hardwareProfiles/useHardwareProfiles';
+
+type HardwareProfileIdentifierMultiselectProps = {
+  data: string[];
+  setData: (data: string[]) => void;
+};
+
+export const HardwareProfileIdentifierMultiselect: React.FC<
+  HardwareProfileIdentifierMultiselectProps
+> = ({ data, setData }) => {
+  const { dashboardNamespace } = useDashboardNamespace();
+  const [hardwareProfiles] = useHardwareProfiles(dashboardNamespace);
+
+  const identifiers = React.useMemo(() => {
+    const uniqueIdentifiers = new Set<string>(data);
```

Comment:
We can add return condition for empty data so that it doesn't go through unnecessary operations like creating a Set and filtering. 
```suggestion
  const identifiers = React.useMemo(() => {
    if (!hardwareProfiles.length && !data.length) return [];
    const uniqueIdentifiers = new Set<string>(data);
```